volatility3/cli/__init__.py

```diff
@@ -792,8 +790,8 @@ def close(self):
                     return None
 
                 self._file.close()
-                output_filename = self._get_final_filename()
-                os.rename(self._name, output_filename)
+                self._output_filename = self._get_final_filename()
```

Comment:
Self variables should be defined (and set to a default value, even if it's `None`) in the object's `__init__` method.

Reply:
I'd almost sooner this sets the preferred filename *just* before the file is actually closed.  I think the intention was that it would act as the filename that had been output, based on https://github.com/volatilityfoundation/volatility3/blob/a66aa020507ec3486ceef5ed73e43614856403d4/volatility3/framework/interfaces/plugins.py#L41.  The comment suggests that this value would be set to the correct filename once the file is closed, could be done on this line instead of creating a new (private) variable.  It looks like we never did that, but that was the intention.